src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -721,7 +721,7 @@ def _validate_vm_vmss_create_public_ip(cmd, namespace):
         from azure.cli.core.profiles import ResourceType
         PublicIPAddressSkuName, IPAllocationMethod = cmd.get_models('PublicIPAddressSkuName', 'IPAllocationMethod',
                                                                     resource_type=ResourceType.MGMT_NETWORK)
-        if namespace.public_ip_sku == PublicIPAddressSkuName.standard.value:
+        if PublicIPAddressSkuName is not None and namespace.public_ip_sku == PublicIPAddressSkuName.standard.value:
```

Comment:
We don't need this check because this is redundant from `getattr(namespace, 'public_ip_sku', None)` 3 lines above. Also related argument is already under api-version condition check. Please let me know if I have missed anything; otherwise, please close this PR and update the test code to skip the scenario if test under stack.
We already have layers of layers conditions made for stack, I really hope we can minimize those.

Reply:
Ok, I will remove the test code. Thanks!